apps/hermes/client/js/src/HermesConnection.ts

```diff
@@ -0,0 +1,198 @@
+   * it will timeout regardless of the retries at the configured `timeout` time.
+   */
+  httpRetries?: number;
+};
+
+export class HermesConnection {
+  private baseURL: string;
+  private timeout: DurationInMs;
+  private httpRetries: number;
+
+  /**
+   * Constructs a new Connection.
+   *
+   * @param endpoint endpoint URL to the price service. Example: https://website/example/
+   * @param config Optional HermesConnectionConfig for custom configurations.
+   */
+  constructor(endpoint: string, config?: HermesConnectionConfig) {
+    this.baseURL = endpoint;
+    this.timeout = config?.timeout || DEFAULT_TIMEOUT;
+    this.httpRetries = config?.httpRetries || DEFAULT_HTTP_RETRIES;
+  }
+
+  private async httpRequest<ResponseData>(
+    url: string,
+    options?: RequestInit,
+    retries = this.httpRetries,
+    backoff = 300
+  ): Promise<ResponseData> {
+    const controller = new AbortController();
+    const { signal } = controller;
+    options = { ...options, signal }; // Merge any existing options with the signal
+
+    // Set a timeout to abort the request if it takes too long
+    const timeout = setTimeout(() => controller.abort(), this.timeout);
+
+    try {
+      const response = await fetch(url, options);
+      clearTimeout(timeout); // Clear the timeout if the request completes in time
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+      return await response.json();
+    } catch (error) {
+      clearTimeout(timeout);
+      if (
+        retries > 0 &&
+        !(error instanceof Error && error.name === "AbortError")
+      ) {
+        // Wait for a backoff period before retrying
+        await new Promise((resolve) => setTimeout(resolve, backoff));
+        return this.httpRequest(url, options, retries - 1, backoff * 2); // Exponential backoff
+      }
+      if (error instanceof Error) {
+        throw error;
+      } else {
+        throw new Error("An unknown error occurred");
+      }
+    }
+  }
+
+  /**
+   * Fetch the set of available price feeds.
+   * This endpoint can be filtered by asset type and query string.
+   * This will throw an error if there is a network problem or the price service returns a non-ok response.
+   *
+   * @param query Optional query string to filter the price feeds. If provided, the results will be filtered to all price feeds whose symbol contains the query string. Query string is case insensitive. Example : bitcoin
+   * @param filter Optional filter string to filter the price feeds. If provided, the results will be filtered by asset type. Possible values are crypto, equity, fx, metal, rates. Filter string is case insensitive. Available values : crypto, fx, equity, metals, rates
+   * @returns Array of PriceFeedMetadata objects.
+   */
+  async getPriceFeeds(
+    query?: string,
+    filter?: string
+  ): Promise<PriceFeedMetadata[]> {
+    const url = new URL(`${this.baseURL}/v2/price_feeds`);
```

Comment:
Nice use of URL but rather than interpolating it's preferable to do `new URL('/v2/price_feeds', this.baseURL)` since doing it that will will enable [relative references to resolve properly](https://developer.mozilla.org/en-US/docs/Web/API/URL_API/Resolving_relative_references)

Reply:
thanks for the tips